src/main/java/de/techdev/trackr/domain/employee/vacation/support/MailApproveService.java

```diff
@@ -0,0 +1,119 @@
+package de.techdev.trackr.domain.employee.vacation.support;
+
+import de.techdev.trackr.domain.common.UuidMapper;
+import de.techdev.trackr.domain.employee.vacation.VacationRequest;
+import de.techdev.trackr.domain.employee.vacation.VacationRequestApproveService;
+import de.techdev.trackr.domain.employee.vacation.VacationRequestRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import java.io.IOException;
+
+@Slf4j
+@Service
+@Profile("gmail")
+public class MailApproveService {
+
+    @Autowired
+    private UuidMapper uuidMapper;
+
+    @Autowired
+    private VacationRequestRepository vacationRequestRepository;
+
+    @Autowired
+    private VacationRequestApproveService vacationRequestApproveService;
+
+    public void approveOrRejectFromMail(Message mail) {
+        log.debug("Got mail.");
+        String subject;
+        String content;
+        String from;
+        MessageWrapper message = new MessageWrapper(mail);
+        try {
+            subject = mail.getSubject();
+            content = message.extractContentAsString();
+            from = message.getSender();
+        } catch (MessagingException | IOException e) {
+            throw new RuntimeException("Could not extract uuid or content from the mail.", e);
+        }
+        String uuid = uuidMapper.extractUUIDFromString(subject);
+
+        if (uuid == null) {
+            throw new RuntimeException("Could find a UUID in the subject " + subject);
+        }
+        log.debug("Extracted UUID {} from subject." , uuid);
+        Long id = uuidMapper.getIdFromUUID(uuid);
+
+        if (id != null) {
+            log.debug("Got ID {} from the mapper.", id);
+            actualApprove(id, from, content);
+        } else {
+            log.debug("Did not find an ID for UUID {}, deleting record.", uuid);
+            uuidMapper.deleteUUID(uuid);
+        }
+    }
+
+    /**
+     * Count how often a search string is present in a text.
+     * @param text The text to analyze.
+     * @param search The search string to find.
+     * @return The number of occurrences of search in text, zero if any of the arguments is null.
+     */
+    protected int containsCount(String text, String search) {
```

Comment:
maybe we can replace this with `org.springframework.util.StringUtils.countOccurrencesOf(text, search);`